### ISSUE TITLE
[BUG] Fails to prevent Freevee ads on Amazon UK

### ISSUE THREAD

**Issue by LumosX** [Dreamlinerm/Netflix-Prime-Auto-Skip#85]
### Describe the bug
The extension does not appear to function on freevee content on Amazon Prime UK.

### To Reproduce
1. Open Amazon Prime UK.
2. Pick any Freevee content: for example, "Falling Skies" season 1 (linked in the background info section)
3. Observe ad "blips" in the seek bar. When an ad triggers, skipping it is impossible and there is no way to fast-forward through it (except for increasing playback speed to 2.0x if the "improve UI" option in the extension is enabled. This, of course, doesn't help).

I can reproduce this with any freevee content I try watching, such as season 3 of "Renegade".

### Expected behaviour
Either the ad is skipped, terminating immediately after starting, or there's a one-click manual solution, such as a "skip" button available during the ad. I dunno, I've never used this extension before so I imagine the expected behaviour is one of these two options.

### Background information:
 - Exact URL(includes show): [Falling Skies season 1](https://www.amazon.co.uk/gp/video/detail/B00IC6JQDQ/ref=atv_hm_hom_c_Wxw9N3_5_1); same problem observed when accessed through "continue watching" [from the main Prime Video page](https://www.amazon.co.uk/Amazon-Video/b?node=3010085031); reproduced using [Renegade season 3](https://www.amazon.co.uk/gp/video/detail/B07RJ7RYYP/ref=atv_hm_hom_c_Wxw9N3_5_1) too
 - Country: United Kingdom
 - Browser: Firefox (121.0.1 and the latest 122.0; updating made no difference)
 - Addon-Version: 1.0.81
 - Console logs: [console-export-2024-1-25_14-37-2.txt](https://github.com/Dreamlinerm/Netflix-Prime-Auto-Skip/files/14053457/console-export-2024-1-25_14-37-2.txt)
To generate the logs, I opened a new episode (Falling Skies s01e06), opened the console, "seeked" (sought?) to right before the ad, watched the ad (the extension did not skip it or permit me to skip/fast-forward it, UI controls were locked out as normal), paused the TV show once I was returned to it, and exported the logs. The ad itself should have occurred right before the last big "uncaught (in promise)" exception, which I believe was generated right as the ad ended.

#### Extra details
The repo README specifies that freevee ads are matched by looking for an `atvwebplayersdk-adtimeindicator-text` class. I took the [following screenshot of the DOM](https://github.com/Dreamlinerm/Netflix-Prime-Auto-Skip/assets/17273782/e8fd8307-cfa0-4e72-a5d9-77cede55678f) during an ad segment: the closest thing we have is `atvwebplayersdk-ad-timer-text`. "Adtimeindicator" is not present anywhere in the model. I'd wager class names might have been changed after this functionality was initially implemented, leading to the extension's apparent inability to capture freevee ads; or perhaps the class name is different for Amazon UK only.

Perhaps matching by `aria-label` would be a better idea? An Amazon dev reading this post right now (or the repo readme later) could easily push a change to refactor class names and break a strict name match, but matching against accessibility features could make the extension more resilient to future changes. Not sure if worth exploring though.

#### NB: extra bug spotted
Additionally, there also seems to be a problem with whatever API the extension uses for TMDB ratings: 
* At the top of the file, we see that the query for "Falling Skies" apparently returns a "no score found" error that appears to instead be looking for some sort of pornography; "Falling Skies" itself has a 71% TMDB rating.
* This is repeated for multiple items, for example "Farscape" returns a "no score found" error with "Farscape Undressed" in the title.
* I see "The Shield" marked with a 4.8 by this extension, whilst TMDB itself reports an 81% rating; this does _not_ produce a "no score found" entry.

#### In conclusion
Thanks for your hard work on this extension, and consider this a low priority bug report. Just wanted to report this on behalf of any brits that might find themselves in a similar position to me one day.

**Comment by Dreamlinerm** [comment]
Amazon is implementing ads accords the board starting the 5. of February I think many things might change at that date.

**Comment by Dreamlinerm** [comment]
They changed the id from atvwebplayersdk-adtimeindicator-text to atvwebplayersdk-ad-timer-text

**Comment by Dreamlinerm** [comment]
I checked my browser, they did the UI change in my country too

**Comment by Dreamlinerm** [comment]
fixed in version 82

**Comment by Dreamlinerm** [comment]
The TMDB stuff I will look into later, but I just use the API, maybe I can filter out which results are correct.

**Comment by LumosX** [comment]
Incredible, cheers mate.

**Comment by Dreamlinerm** [comment]
The adTimeindicator does not have a aria-label otherwise I would have used it yes.

**Comment by Dreamlinerm** [comment]
The no Score found logs I cannot fix, except the movie title was not caught correctly e.g. there is season 1 included in the title or sth. like that. The score is incorrect I will look into

**Comment by Dreamlinerm** [comment]
The TMDB website search is way better than the API search. That's what I already found out.

**Comment by Dreamlinerm** [comment]
I looked at the TMDB APIs, and I was using movie before, but I think the multi API fits better and I excluded adult ratings.
These are all the APIs.
The wrong Ratings will refresh every month since there is a Cache built into my add-on.
https://developer.themoviedb.org/reference/search-multi 

Thank you for writing about the adult shows I excluded that now, and I did not realize that tv shows were not included in this api.

**Comment by Dreamlinerm** [comment]
I still have to program if its the correct movie tho, but thats very difficult since every movie from the api basically has the title included
